app/models/effective/post.rb

```diff
@@ -67,6 +70,10 @@ class Post < ActiveRecord::Base
       scope
     }
 
+    def to_param
+      slug || "#{id}-#{title.parameterize}"
+    end
+
```

Comment:
I don't know if having this fallback is a good idea, but it sure helps with the old data.